components/launcher-protocol/src/types.rs

```diff
@@ -276,3 +276,47 @@ impl LauncherMessage for Shutdown {
 impl From<Shutdown> for generated::Shutdown {
     fn from(_value: Shutdown) -> Self { generated::Shutdown {} }
 }
+
+#[derive(Clone, Debug)]
+pub struct PidOf {
+    pub service_name: String,
+}
+
+impl LauncherMessage for PidOf {
+    type Generated = generated::PidOf;
+
+    const MESSAGE_ID: &'static str = "PidOf";
+
+    fn from_proto(proto: generated::PidOf) -> Result<Self> {
+        Ok(PidOf { service_name: proto.service_name
+                                      .ok_or(Error::ProtocolMismatch("service_name"))?, })
+    }
+}
+
+impl From<PidOf> for generated::PidOf {
+    fn from(value: PidOf) -> Self { generated::PidOf { service_name: Some(value.service_name), } }
+}
+
+#[derive(Clone, Debug)]
+pub struct PidIs {
+    pub pid: Option<u32>,
+}
+
+impl LauncherMessage for PidIs {
+    type Generated = generated::PidIs;
+
+    const MESSAGE_ID: &'static str = "PidIs";
+
+    fn from_proto(proto: generated::PidIs) -> Result<Self> {
+        // TODO (CM): ensure that the Pid is never Some(0)
+        Ok(PidIs { pid: proto.pid })
+    }
+}
+
+impl From<PidIs> for generated::PidIs {
+    // TODO (CM): I would need to ensure that PidIs can never contain
```

Comment:
Could you use [`std::num::NonZeroU32`](https://doc.rust-lang.org/std/num/struct.NonZeroU32.html)?

Reply:
Yeah, that's kinda what I'm thinking. Hopefully that's not a huge refactor, but something I'm thinking about for a follow-on PR.